geopandas/tools/clip.py

```diff
@@ -7,33 +7,39 @@
 """
 import warnings
 
-from shapely.geometry import Polygon, MultiPolygon
+from shapely.geometry import Polygon, MultiPolygon, box
 
 from geopandas import GeoDataFrame, GeoSeries
 from geopandas.array import _check_crs, _crs_mismatch_warn
 
 
-def _clip_gdf_with_polygon(gdf, poly):
-    """Clip geometry to the polygon extent.
+def _clip_gdf_with_mask(gdf, mask):
+    """Clip geometry to the polygon/rectangle extent.
 
-    Clip an input GeoDataFrame to the polygon extent of the poly
+    Clip an input GeoDataFrame to the polygon extent of the polygon
     parameter.
 
     Parameters
     ----------
     gdf : GeoDataFrame, GeoSeries
         Dataframe to clip.
 
-    poly : (Multi)Polygon
-        Reference polygon for clipping.
+    mask : (Multi)Polygon, tuple
+        Reference polygon/rectangle for clipping.
 
     Returns
     -------
     GeoDataFrame
         The returned GeoDataFrame is a clipped subset of gdf
-        that intersects with poly.
+        that intersects with polygon/rectangle.
     """
-    gdf_sub = gdf.iloc[gdf.sindex.query(poly, predicate="intersects")]
+    clipping_by_rectangle = isinstance(mask, tuple)
+
+    intersection_polygon = mask
+    if clipping_by_rectangle:
+        intersection_polygon = box(*mask)
```

Comment:
```suggestion
    if clipping_by_rectangle:
        intersection_polygon = box(*mask)
    else:
         intersection_polygon = mask
```

I know that this is essentially the same thing so feel free to discard this. I just think that having a complete if/else makes it a bit more readable.

Reply:
Done